examples/next-openai/app/tool-calling/page.tsx

```diff
@@ -0,0 +1,98 @@
+'use client';
+
+import { ChatRequest, ToolCallHandler, nanoid } from 'ai';
+import { Message, useChat } from 'ai/react';
+
+export default function Chat() {
+  const toolCallHandler: ToolCallHandler = async (chatMessages, toolCalls) => {
+    let handledFunction = false;
+    for (const tool of toolCalls) {
+      if (tool.type === 'function') {
+        const result = handleFunction(tool.function);
```

Comment:
Is there a possibility I don't want the function to return anything and just perform some side-effect? We may need to set handledFunction / push a message when we call `handleFunction` regardless of the return value. Maybe we recommend try/catch and throwing an error instead of returning one.

Reply:
`handleFunction` is client code (see below) - not sure I fully follow. Users should be able to do what they want. I'll see if I can simplify the code (copied from Svelte for consistency).

Reply:
We're patiently waiting for this fix to be released. It seems like the only thing holding it up is an example implementation? Can we get this merged so we can start using the useChat react hook with tools?